data/txntest/txn.go

```diff
@@ -256,8 +256,8 @@ func (tx Txn) Txn() transactions.Transaction {
 			OnCompletion:      tx.OnCompletion,
 			ApplicationArgs:   tx.ApplicationArgs,
 			Accounts:          tx.Accounts,
-			ForeignApps:       tx.ForeignApps,
-			ForeignAssets:     tx.ForeignAssets,
+			ForeignApps:       append([]basics.AppIndex(nil), tx.ForeignApps...),
+			ForeignAssets:     append([]basics.AssetIndex(nil), tx.ForeignAssets...),
```

Comment:
why are you appending yourself back to a nil slice?

Reply:
I want to ensure that, even if the tester writes the empty slice
```
ForeignAssets: []basics.AssetIndex{}
```
that we actually use
```
ForeignAssets: nil
```
because that's what happens when a real transaction is deserialized. (and it breaks some of the tests comparing the two ledgers in DoubleLedger to use two different representations).

The thinking is that testers (including me!) sometimes use the empty slice instead of nil and the error that pops out is _very_ hard to understand.

Reply:
ah, didn't see it coming, but you are def right.  I didn't take serialization of txn into consideration.